R/domainEnrichment.R

```diff
@@ -0,0 +1,164 @@
+# domainEnrichment.R
+
+#' Domain enrichment in a desired system.
+#'
+#' \code{domainEnrichment} Find the enriched domains in selected system and interpret enrichment by plot.
+#'
+#' For the PHYLA system, see details in the \href{../doc/domainEnrichment-Vignette.html}{PHALY domain enrichemnt vignette} (or load the vignette with \code{vignette("domainEnrichment-Vignette", package = "BCB420.2019.ESA")}).
+#'
+#' @section Details: This function is created as part of BCB420 2019 winter course. The domain databaase were fetched from InterPro, see more details in \code{\link{fetchData}}.
+#'
+#' @param system The shortehand notation of selected system, five-letter code as discussed in class.
+#' @param alpha The alpha value for Benjamini-Hochberg control. Default is 0.05.
+#' @return A data frame that include domains in the system with its counts and frequency in system and in random genes; p-value, odd ratio, adjust p-value and enrichment; interpro descriptions. Return NULL if system is not available by \code{\link{SyDBgetSysSymbols}}.
+#'
+#' @importFrom stats p.adjust fisher.test reorder
+#' @import biomaRt
+#' @import ggplot2
+#' @import ggiraph
+#'
+#' @author \href{https://orcid.org/0000-0001-8720-5874}{Fan Shen} (aut)
+#'
+#' @seealso \code{\link{fetchData}} Fetch data from package
+#' @seealso \code{\link{SyDBgetRootSysIDs}} Get the system ID in database
+#' @seealso \code{\link{SyDBgetSysSymbols}} Get the genes in desired system
+#'
+#' @examples
+#' # Plot enrichend domains in PHALY system orderd by rich factor
+#' # Plot would separate to two parts if too many domains are enriched
+#' # User can hover around the point on plot to get more info (i.e ID descripton)
+#' system <- "PHALY"
+#' domainEnrichment(system)
+#'
+#' @export
+domainEnrichment <- function(system, alpha = 0.05) {
+  # 1. Fetch data from Dr. Steipe's BCB420.2019.ESA package
+  # Two lists from InterPro database
```

Comment:
Fetching data inside the function is not a good idea. Better to pass it as a parameter, which can default to fetching it, if it doesn't already exist in the workspace.